xcdat/dataset.py

```diff
@@ -162,145 +176,130 @@ def open_mfdataset(
 
     Keep a single variable in the Dataset:
 
-    >>> from xcdat.dataset import open_dataset
+    >>> from xcdat.dataset import open_mfdataset
     >>> ds = open_mfdataset(["file_path1", "file_path2"], data_var="tas")
-
-    Keep multiple variables in the Dataset:
-
-    >>> from xcdat.dataset import open_dataset
-    >>> ds = open_mfdataset(["file_path1", "file_path2"], data_var=["ts", "tas"])
     """
-    # check if time axis is cf_compliant
-    cf_compliant = _check_dataset_for_cf_compliant_time(paths)
-
-    # if cf_compliant, let xarray decode the time units
-    # otherwise, decode using decode_time_units
-    if cf_compliant:
-        ds = xr.open_mfdataset(paths, decode_times=True, data_vars=data_vars, **kwargs)
-    else:
-        ds = xr.open_mfdataset(paths, decode_times=False, data_vars=data_vars, **kwargs)
-        if ds.cf.dims.get("T") is not None:
-            ds = decode_time_units(ds)
-
+    if decode_times:
+        cf_compliant_time: Optional[bool] = has_cf_compliant_time(paths)
+        # XCDAT handles decoding decoding time values with non-CF units using
+        # the preprocess kwarg.
+        if cf_compliant_time is False:
+            decode_times = False
+            preprocess = partial(_preprocess_non_cf_dataset, callable=preprocess)
```

Comment:
The `xcdat.open_mfdataset()` `preprocess` kwarg gets passed to `_preprocess_non_cf_dataset()`, which then gets passed to the `xr.open_mfdataset()` `preprocess` kwarg.

Reply:
This allows for chaining multiple preprocessing functions.